.ci/scripts/go-plus/gcb-terraform-vcr-tester/test_terraform_vcr.sh

```diff
@@ -21,9 +22,10 @@ pushd $local_path
 
 # Only skip tests if we can tell for sure that no go files were changed
 echo "Checking for modified go files"
+git fetch origin $old_branch:$old_branch --depth 1
 # get the names of changed files and look for go files
 # (ignoring "no matches found" errors from grep)
-gofiles=$(git diff --name-only HEAD~1 | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" || test $? = 1; })
+gofiles=$(git diff $new_branch $old_branch --name-only | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" || test $? = 1; })
```

Comment:
```suggestion
# If there was no code generated, this will always return nothing (because there's no diff)
gofiles=$(git diff $new_branch $old_branch --name-only | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" || test $? = 1; })
```

Reply:
One nitpick would be that we could structure the "was there code generated" check as a guard clause for a little more clarity- that way after a certain point we definitely know that we generated some code. Commenting works too, though.